src/main/java/com/exactpro/th2/crawler/Crawler.java

```diff
@@ -165,57 +183,78 @@ public Duration process() throws IOException, UnexpectedDataProcessorException {
             intervalStartForProcessor(dataProcessor, interval, state);
 
             Continuation continuation = state == null ? null : typeStrategy.continuationFromState(state);
+
+            // At first, we need to process messages for all session aliases we have.
             DataParameters parameters = new DataParameters(info, crawlerId, sessionAliases);
-            CrawlerData<Continuation> data;
-            Report<Continuation> sendingReport = null;
-            long processedEvents = 0L;
-            Timestamp startTime = toTimestamp(interval.getStartTime());
-            Timestamp endTime = toTimestamp(interval.getEndTime());
+            LOGGER.debug("Crawler is going to process messages with {} aliases: {}. Interval from {} to {}",
+                    sessionAliases.size(), String.join(", ", sessionAliases), interval.getStartTime(), interval.getEndTime());
+
+            Collection<String> newAliases;
+
             do {
```

Comment:
I think we should check if the aliases are not empty before making any requests. Because if aliases are empty there is no point to make any request to data provider

Reply:
I'm not sure if I have understood you correctly, can you please check out my last commit?

Reply:
I meant we should check the aliases retrieved from the data provider. The parameters verification should be done earlier in the constructor.